pallets/ddc-verification/src/lib.rs

```diff
@@ -9,21 +9,64 @@
 #![recursion_limit = "256"]
 
 use ddc_primitives::{ClusterId, DdcEra, MmrRootHash};
-use frame_support::pallet_prelude::*;
-use frame_system::pallet_prelude::*;
+use frame_support::{pallet_prelude::*, traits::OneSessionHandler};
+use frame_system::{
+	offchain::{AppCrypto, CreateSignedTransaction, SendSignedTransaction, Signer},
+	pallet_prelude::*,
+};
 pub use pallet::*;
+use sp_application_crypto::RuntimeAppPublic;
+use sp_core::crypto::KeyTypeId;
+use sp_runtime::Percent;
 use sp_std::prelude::*;
 
 pub mod weights;
 use crate::weights::WeightInfo;
 
-#[cfg(feature = "runtime-benchmarks")]
-pub mod benchmarking;
+type BatchIndex = u16;
+pub const KEY_TYPE: KeyTypeId = KeyTypeId(*b"cer!");
+pub mod sr25519 {
+	mod app_sr25519 {
+		use sp_application_crypto::{app_crypto, sr25519};
 
-#[cfg(test)]
-pub(crate) mod mock;
-#[cfg(test)]
-mod tests;
+		use crate::KEY_TYPE;
+		app_crypto!(sr25519, KEY_TYPE);
+	}
+
+	sp_application_crypto::with_pair! {
+		pub type AuthorityPair = app_sr25519::Pair;
+	}
+	pub type AuthoritySignature = app_sr25519::Signature;
+	pub type AuthorityId = app_sr25519::Public;
+}
+pub mod crypto {
+	use sp_core::sr25519::Signature as Sr25519Signature;
+	use sp_runtime::{
+		app_crypto::{app_crypto, sr25519},
+		traits::Verify,
+		MultiSignature, MultiSigner,
+	};
+
+	use super::KEY_TYPE;
+	app_crypto!(sr25519, KEY_TYPE);
+
+	pub struct OffchainIdentifierId;
+
+	impl frame_system::offchain::AppCrypto<MultiSigner, MultiSignature> for OffchainIdentifierId {
+		type RuntimeAppPublic = Public;
+		type GenericSignature = sp_core::sr25519::Signature;
+		type GenericPublic = sp_core::sr25519::Public;
+	}
+
+	// implemented for mock runtime in test
+	impl frame_system::offchain::AppCrypto<<Sr25519Signature as Verify>::Signer, Sr25519Signature>
```

Comment:
can we move Sr25519 to runtime configuraiton, so that the pallet doesn't know which exact algorithm is used?